### ISSUE TITLE
Pub/Sub PullRequest.maxMessages is mandatory

### ISSUE THREAD

**Issue by mziccard** [GoogleCloudPlatform/gcloud-java#997]
The `PullRequest.maxMessages` is a mandatory field, if we don't provide it we get this error:
```json
{
 "error": {
  "code": 400,
  "message": "A required argument is missing in the request: (argument=\"max_messages\").",
  "status": "INVALID_ARGUMENT",
 }
}
```

The PubSub API design we drafted considers `maxMessages` an optional option, we must either fix this (by transforming it into a mandatory parameter) or use a default value if the user does not explicitly provide it. /cc @eschapira @aozarov thoughts?

**Comment by eschapira** [comment]
Is this for calling the pull method explicitly or for the callback version?
I would say that for explicit pulls it is required, and for callback there
should be a default (set it to 100, no hard data on why that's a good
value, but sounds reasonable to me).

On Mon, May 9, 2016 at 9:52 AM, Marco Ziccardi <notifications@github.com>
wrote:

> The PullRequest.maxMessages is a mandatory field, if we don't provide it
> we get this error:
>
> {
>  "error": {
>   "code": 400,
>   "message": "A required argument is missing in the request: (argument=\"max_messages\").",
>   "status": "INVALID_ARGUMENT",
>  }
> }
>
> The PubSub API design we drafted considers maxMessages an optional
> option, we must either fix this (by transforming it into a mandatory
> parameter) or use a default value if the user does not explicitly provide
> it. /cc @eschapira <https://github.com/eschapira> @aozarov
> <https://github.com/aozarov> thoughts?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-java/issues/997>
>



-- 
Emilio Schapira

**Comment by mziccard** [comment]
Yes, this is for calling the pull method explicitly.

For the pull method with callback the idea was to allow the user to limit
the maximum number of concurrent callbacks. 100 sounds like a reasonable
value in this case.

**Comment by mziccard** [comment]
Fixed in #996
